firmware/shared/src/hw/hw_can.c

```diff
@@ -87,17 +98,21 @@ bool io_can_transmit(const CanHandle *can_handle, CanMsg *msg)
     tx_header.TransmitGlobalTime = DISABLE;
 
     // Spin until a TX mailbox becomes available.
-    while (HAL_CAN_GetTxMailboxesFreeLevel(can_handle->hcan) == 0U)
+    static const uint16_t TRIES = 1000; // empirically, makes it try for about 1ms
+    for (int i = 0; i < TRIES && HAL_CAN_GetTxMailboxesFreeLevel(can_handle->hcan) == 0U; i++)
         ;
+    if (HAL_CAN_GetTxMailboxesFreeLevel(can_handle->hcan) == 0U)
+        return false;
```

Comment:
Do we need this? If we do we should probably think harder about the timeout to avoid potentially dropping messages during high bus load

Reply:
if a single message is taking more than 1ms then you are in a situation where you can only send 10 messages per 100hz task. i think you're probably cooked anyways in that case. i am happy to bump this up to like 5ms but otherwise, I think we should start clearing stale messages from the queue.

Reply:
Hmm in my opinion this doesn't really solve that problem? If If our messages are regularly taking this long then yeah we're probably cooked. Either we pop them off the front due to a timeout, or the CAN queues fill up and we get a warning. The advantage of letting them fill up is we know when this happens due to the warning.

Also I think this does add a risk that we configure the TRIES var wrong then we end up with dropped messages during moments of high bus load. I honestly don't know what the maximum latency on adding a messages is, have you checked on the car or anything like that? 

Seems like this doesn't really help solve any problems (and potentially hides it if this means the warning doesn't get set), and adds a potential risk which would be tricky to debug.